src/main/java/com/ericsson/gerrit/plugins/eiffel/events/generators/EiffelSourceChangeSubmittedEventGenerator.java

```diff
@@ -70,22 +64,10 @@ public static EiffelSourceChangeSubmittedEvent generate(final EiffelPluginConfig
         eiffelEvent.eventParams.data.gitIdentifier.branch = branch;
         eiffelEvent.eventParams.data.gitIdentifier.repoName = projectName;
 
-        final String previousSourceChangeCreatedEvent = getPreviousEiffelEventId(pluginConfig,
-                EventType.SCC_EVENT, changeId);
-        final Link changeLink = createLink(LINK_TYPE_CHANGE, previousSourceChangeCreatedEvent);
-        if (changeLink != null) {
-            eiffelEvent.eventParams.links.add(changeLink);
-        }
-
-        final List<String> parentsSHAs = commitInformation.getParentsSHAs(commitId, projectName);
-        final String previousSourceChangeSubmittedEventId = getPreviousEiffelEventId(pluginConfig,
-                EventType.SCS_EVENT,
-                parentsSHAs);
-        final Link previousVersionLink = createLink(LINK_TYPE_PREVIOUS_VERSION,
-                previousSourceChangeSubmittedEventId);
-        if (previousVersionLink != null) {
-            eiffelEvent.eventParams.links.add(previousVersionLink);
-        }
+        final LinkGenerator linkGenerator = new LinkGenerator(pluginConfig, commitInformation);
+        linkGenerator.addScsChange(changeId);
```

Comment:
If we move this logic to the constructor then we can avoid having to call it in every event.

Reply:
What constructor?

Reply:
The two events have different links though so the constructor would not work for both. There would also be an increase of parameters passed.

Reply:
ok